File: 168-excel-sheet-column-title/excel-sheet-column-title.cpp
class Solution {
public:
    std::string convertToTitle(int n) {
        if (n < 27) {
            return std::string(1, 'A' + (n - 1) % 26);
        }
        
        std::string c = "";
        while (n > 0) {
            if (n % 26 == 0) {
                c += 'A' + 25;
                n -= 1;
            } else {
                c += 'A' + n % 26 - 1;
            }
            n /= 26;
        }
        std::reverse(c.begin(), c.end());
        return c;
    }
};

File: 168-excel-sheet-column-title/excel-sheet-column-title.java
class Solution {
    public String convertToTitle(int c) {
        StringBuilder sol = new StringBuilder();
        while (c > 0) {
            int temp = (c - 1) % 26; 
            sol.append((char) ('A' + temp));
            c = (c - 1) / 26;
        }
        return sol.reverse().toString(); 
    }
}